### PR TITLE
addingCars

### PR DESCRIPTION
**Pull request by eduardgasparyan**
Cars added

### CODE REVIEW

File: routers/userRouting.js [hunk 1]
```diff
@@ -49,4 +50,46 @@ router.get(`/logout/:username`, async ctx => {
   return au.logout(data);
 });
 
+router.post(`/:username/addCar`, async ctx => {
+  const data = {
+    username: ctx.params,
+    bodyParsed: ctx.request.body,
+    accessToken: ctx.request.headers.authorization,
+  };
+  if (await au.verifyToken(data.accessToken, data.username))
```
Comment (Karlen92) [inline routers/userRouting.js]: you need to have a middleware, and do this in there, and not call every time

File: services/userCars.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+const sequelize = require('../models');
+
+exports.addingCar = async data => {
```
Comment (Karlen92) [inline services/userCars.js]: addCar not addingCar

File: services/userCars.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const user = await sequelize.User.findOne({
+    where: {
+      username: data.username.username,
+    },
+  });
+  const databaseCarCheck = await sequelize.car.findOne({
+    where: {
+      car: data.bodyParsed.car,
```
Comment (Karlen92) [inline services/userCars.js]: why do you have the name `bodyParsed` ?

File: routers/userRouting.js [hunk 4]
```diff
@@ -49,4 +50,46 @@ router.get(`/logout/:username`, async ctx => {
...
+    username: ctx.params,
+    accessToken: ctx.request.headers.authorization,
+    carName: ctx.request.body,
+  };
+  if (await au.verifyToken(data.accessToken, data.username))
+    return car.deleteCar(data);
+  else throw new Error('Verification false');
+});
+
+router.post(`/:username/changeCar`, async ctx => {
```
Comment (Karlen92) [inline routers/userRouting.js]: this should be with usedId

File: routers/userRouting.js [hunk 5]
```diff
@@ -49,4 +50,46 @@ router.get(`/logout/:username`, async ctx => {
...
+  const data = {
+    username: ctx.params,
```
Comment (Karlen92) [inline routers/userRouting.js]: `ctx.params` is not a `username`, make sure to take only what you need

File: routers/userRouting.js [hunk 6]
```diff
@@ -49,4 +50,46 @@ router.get(`/logout/:username`, async ctx => {
...
+    accessToken: ctx.request.headers.authorization,
+    carName: ctx.request.body,
+  };
+  if (await au.verifyToken(data.accessToken, data.username))
```
Comment (Karlen92) [inline routers/userRouting.js]: if else block should be always like this

```
if(){
}else {
}
```

File: routers/userRouting.js (hunk 4 above)
Comment (Karlen92) [inline routers/userRouting.js]: Every user can change any user data, you don't have authorization here

**Review by Karlen92 — CHANGES_REQUESTED** [review]

File: routers/userRouting.js (hunk 4 above)
Comment (eduardgasparyan) [reply routers/userRouting.js]: It means, that anyone can change another ones cars?

Comment (eduardgasparyan) [reply routers/userRouting.js]: What should be with userId?

**Pull request merged**